test/swar/BasicOperations.cpp

```diff
@@ -565,10 +583,188 @@ TEST_CASE(
+        CHECK(S4_32(0x0008'0080).value() == firstZeroLane<4, uint32_t>( S4_32(0xaaa0'aa0a)));
+        CHECK(1 == S4_32(firstZeroLane<4, uint32_t>( S4_32(0xaaa0'aa0a))).lsbIndex());
+        CHECK(S4_32(0x8888'8888).value() == firstZeroLane<4, uint32_t>( S4_32(0x1110'1110)));
+        CHECK(S4_32(0x8800'0000).value() == firstZeroLane<4, uint32_t>( S4_32(0x1025'3552)));
+        CHECK(S4_32(0x8880'0000).value() == firstZeroLane<4, uint32_t>( S4_32(0x1105'5352)));
+
+        // firstZeroLane finds the least significant zero (little endian). This
+        // results in the least significant zero being marked, and everything more
+        // significant than it being turned into garbage data.
+        const char * c = "aaa\0aaa\0";
+        uint64_t x;
+        memcpy(&x, c, 8);
+        S8_64 testBytes{x};
+        auto nulls = firstZeroLane<8, uint64_t>(testBytes);
+        CHECK(3 == S8_64(nulls).lsbIndex());
+    }
+
+    {
+        CHECK(S4_32(0x0000'0000).value() == firstMatchingLane<4, uint32_t>(S4_32(0x7321'4257),
+                                                                           S4_32(0x0000'0000)));
+        CHECK(S4_32(0x0000'8000).value() == firstMatchingLane<4, uint32_t>(S4_32(0x7325'0257),
+                                                                           S4_32(0x0000'0000)));
+        CHECK(S4_32(0x0000'0000).value() == firstMatchingLane<4, uint32_t>(S4_32(0x7341'4157),
+                                                                           S4_32(0x2222'2222)));
+        CHECK(S4_32(0x0880'0800).value() == firstMatchingLane<4, uint32_t>(S4_32(0x7321'4257), 
+                                                                           S4_32(0x2222'2222)));
+        CHECK(S4_32(0x0880'0800).value() == firstMatchingLane<4, uint32_t>(S4_32(0x7321'4357), 
+                                                                           S4_32(0x2222'3333)));
+    }
+
+}
+
+TEST_CASE(
+    "equals_S4_16",
+    "[swar]"
+) {
+    // This test is intentionally horrific b/c blit / at / etc do not work right for 16 bit SWARs.
+    auto incr_lane_zero = [](S4_16 s, int i)  ->S4_16{
+        return s + S4_16{1};
+    };
+    auto incr_lane_one = [](S4_16 s, int i)  ->S4_16{
+        return s + S4_16{0x0010};
+    };
+    auto incr_lane_two = [](S4_16 s, int i)  ->S4_16{
+        return s + S4_16{0x0100};
+    };
+    auto incr_lane_three = [](S4_16 s, int i)  ->S4_16{
+        return s + S4_16{0x1000};
+    };
+    auto cmp = [](S4_16 test, S4_16 expected, int i) ->S4_16 {return equals(test, expected);};
+    auto expectedf_zero = [](S4_16 in, int i) ->S4_16 {return S4_16{S4_16::MostSignificantBit}.clear(0);};
+    auto expectedf_one = [](S4_16 in, int i) ->S4_16 {return S4_16{S4_16::MostSignificantBit}.clear(1);};
+    auto expectedf_two = [](S4_16 in, int i) ->S4_16 {return S4_16{S4_16::MostSignificantBit}.clear(2);};
+    auto expectedf_three = [](S4_16 in, int i) ->S4_16 {return S4_16{S4_16::MostSignificantBit}.clear(3);};
```

Comment:
Why the explicit typing?
This copy and paste deserves preprocessing, we gain nothing copy and pasting this except facilitating superficial reading, at the cost of missing the point of what's really going on.

Reply:
'horrifying unit test left in because S4_16 is broken'

Reply:
I do not understand the response.

Reply:
Something is wrong when using .at() and similar with uint16_t underlying type. as the comment at the top of the block says, I'm leaving this test sort of obviously wretched looking to remind us that we need to figure out why uint16_t doesn't work well as a base type for SWAR.